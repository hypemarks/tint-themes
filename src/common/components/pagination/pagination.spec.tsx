import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import Pagination from './pagination';

const onNewContent = jest.fn();

describe('Test <Pagination /> Component', () => {
  it('should render snapshot', () => {
    const fetchPrevPage = jest.fn();
    const fetchNextPage = jest.fn();

    const wrapper = render(
      <Pagination onNewContent={onNewContent} fetchPrevPage={fetchPrevPage} fetchNextPage={fetchNextPage} />
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should call prev page', () => {
    const fetchPrevPage = jest.fn();
    const fetchNextPage = jest.fn();
    const wrapper = render(
      <Pagination onNewContent={onNewContent} fetchPrevPage={fetchPrevPage} fetchNextPage={fetchNextPage} />
    );
    const paginationFirstButton = wrapper.getByText('Prev');
    fireEvent.click(paginationFirstButton);
    expect(fetchPrevPage).toBeCalled();
  });

  it('should call next page', () => {
    const fetchPrevPage = jest.fn();
    const fetchNextPage = jest.fn();
    const wrapper = render(
      <Pagination onNewContent={onNewContent} fetchPrevPage={fetchPrevPage} fetchNextPage={fetchNextPage} />
    );
    const paginationNextButton = wrapper.getByText('Next');
    fireEvent.click(paginationNextButton);
    expect(fetchNextPage).toBeCalled();
  });
});
