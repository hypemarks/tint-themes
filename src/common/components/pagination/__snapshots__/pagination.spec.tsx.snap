// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Pagination /> Component should render snapshot 1`] = `
<DocumentFragment>
  <button
    class="pagination button"
    data-testid="button"
    tabindex="0"
    type="button"
  >
    Prev
  </button>
  <button
    class="pagination button"
    data-testid="button"
    tabindex="0"
    type="button"
  >
    Next
  </button>
</DocumentFragment>
`;
